### ISSUE TITLE
[Submission] GNU Bourne Again SHell for AArch64

### ISSUE THREAD

**Issue by ianmacd** [Magisk-Modules-Repo/submission#229]
https://github.com/ianmacd/bash-aarch64

**Comment by yochananmarqos** [comment]
Thanks, but we already have [Bash For Android](https://github.com/Magisk-Modules-Repo/bash) and [Terminal Modifications](https://github.com/Magisk-Modules-Repo/terminalmods).

**Comment by MagiskBot** [comment]
Submission rejected

**Comment by ianmacd** [comment]
The existing Bash module isn't for 64 bit ARM, and no part of this new module was copied from that one.

**Comment by Zackptg5** [comment]
@ianmacd 
arm bash works fine on arm64 but this is the one thing I've been wanting to get working for a while. 
So you use gcc instead of android ndk and it worked with it? Interesting.
Have you tried the patches I found from around the web [here? ](https://github.com/Zackptg5/Bash-for-Android-Build-Script/tree/master/patches_5)

I'd be happy to add you as a contributor to my existing module if you'd like.
I can handle the unity template and you can handle the bash itself since you're probably more experienced than me with it

**Comment by ianmacd** [comment]
Yes, no funny business was needed to compile: I just downloaded Linaro gcc, added it to my $PATH and built bash.

I haven't used any of your patches. I hadn't even seen them, actually. I just looked at them to see what they do. Some of them aren't needed when building the way I built, but the ones that adapt the shell to cope with the deviant Android environment seem useful, although I wouldn't personally place /sbin ahead of any /system bin directories, since the latter are trivially overridden with Magisk modules and an unprivileged user could reasonably expect the shell to find user binaries before those intended for the system administrator. The root user, on the other hand, **should** have /sbin ahead of any user-modifiable path. It would be a security risk not to.

All of this is a little contrived on a mobile device, but since some people user their device in a multi-user configuration, the same principles could be said to apply.

I built the pristine source after applying the official 001 and 002 patches to 5.0, so the version in this rejected module is **5.0.2(1)-release (aarch64-unknown-linux-gnu)**.

What I really wanted to do was build a static **zsh**, but that is a much more awkward affair.
